manual/manual/html_processing/src/common.ml

```diff
@@ -0,0 +1,123 @@
+open Soup
+open Printf
+
+let debug = Array.to_list Sys.argv
+            |> List.mem "quiet"
+            |> not
+
+let pr = if debug then print_endline else fun _ -> ()
```

Comment:
I didn't have in mind to write a powerful debug function for my basic debug messages, but I'm ok to use your suggestion. How do I acknowledge you for this?

Reply:
No acknowledgement needed. I'd heartily recommend using one like this in general, though - you win the second you have more than one variable to print, and it can also occasionally be a debugging nightmare when the messages aren't being flushed immediately!

Reply:
What would be the difference with this:
```ocaml
let dbg s = Printf.ksprintf (fun s -> if debug then print_endline s) s;
```
(I guess: performance?)

Reply:
Not sure about performance per se, but it's superior when you start using `%a` - the key thing with `ikfprintf` (and `ifprintf`) is that it doesn't actually format the string. You also get to stay in the `fprintf` class of `%a` functions (so `out_channel -> 'a -> unit` rather than `unit -> 'a -> string`).

This does nothing and returns immediately:
```ocaml
Printf.ifprintf stdout "%a" (fun _ _ -> Unix.sleep 5) 42
```
this "does the same nothing", but takes 5 seconds:
```ocaml
Printf.ksprintf ignore "%a" (fun _ _ -> Unix.sleep 5; "") 42
```
That's useful if, for example, what you want to print in the debug statement is computationally intensive, but remember that even `string_of_int` (i.e. `%d`) is not free - `Printf.ikfprintf stdout "%d" 42` doesn't call `string_of_int` (and so also doesn't allocate any string to hold the result)

Reply:
works great, thanks